force_wfmanager/ui/review/plot.py

```diff
@@ -288,17 +288,16 @@ def _update_plot_x_axis(self):
     def _update_plot_x_data(self):
         """ Update data points displayed by the x axis.
         Sets the x-`self._plot_data` to corresponding data in the
-        `self.data_arrays`.
+        `self.analysis_model`.
         This method is called by the `_update_plot` method during
         the callback update.
         This method is called when the `x` axis is changed.
         """
-        if self.x == "" or len(self.data_arrays) == 0:
+        if self.x == "" or self.analysis_model.is_empty:
             self._plot_data.set_data("x", [])
         else:
             self._plot.x_axis.title = self.x
-            x_index = self.analysis_model.value_names.index(self.x)
-            self._plot_data.set_data("x", self.data_arrays[x_index])
+            self._plot_data.set_data("x", self.analysis_model.column(self.x))
```

Comment:
So presumably removing the `on_trait_change` handlers from both `_update_plot_x_axis` and `_update_plot_y_axis` methods, in exchange for:

```
@on_trait_change("x,y")
def _update_plot_axis(self):
    self._update_plot_x_axis()
    self._update_plot_y_axis()
```

would mean that the number of data points displayed should be consistent between axis (as long as the `analysis_model` is not updated between calls to obtain the x and y data).

Reply:
Yes, that would solve the problem, thank you. My only concern here is that we would need to use `@on_trait_change("x,y,color_by")` for the color plot, which essentially is a bad design of handling multiple axis in general.

Reply:
Shall we leave this for future PRs? This is definitely a UI problem that has to be resolved. At the same time, it has a predictable behaviour, and doesn't break things badly.

Reply:
Agreed - let's raise an issue for this as it is a small UX 'bug'

Reply:
Opened an issue #367